src/main/kotlin/gg/skytils/skytilsmod/features/impl/events/TechnoMayor.kt

```diff
@@ -20,62 +20,72 @@ package gg.skytils.skytilsmod.features.impl.events
 import gg.skytils.skytilsmod.Skytils
 import gg.skytils.skytilsmod.Skytils.Companion.mc
-import gg.skytils.skytilsmod.events.impl.PacketEvent.ReceiveEvent
 import gg.skytils.skytilsmod.utils.*
 import net.minecraft.client.renderer.GlStateManager
 import net.minecraft.entity.EntityLivingBase
 import net.minecraft.entity.item.EntityArmorStand
-import net.minecraft.network.play.server.S02PacketChat
+import net.minecraft.entity.passive.EntityPig
 import net.minecraft.util.AxisAlignedBB
 import net.minecraft.util.BlockPos
 import net.minecraft.util.Vec3
 import net.minecraftforge.client.event.RenderLivingEvent
 import net.minecraftforge.client.event.RenderWorldLastEvent
+import net.minecraftforge.event.entity.player.AttackEntityEvent
+import net.minecraftforge.event.entity.player.EntityInteractEvent
 import net.minecraftforge.event.world.WorldEvent
-import net.minecraftforge.fml.common.eventhandler.EventPriority
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
 import java.awt.Color
-import java.util.concurrent.CopyOnWriteArrayList
 
 object TechnoMayor {
-    private val orbLocations = CopyOnWriteArrayList<Vec3>()
+    private val shinyPigs = HashMap<Vec3, EntityPig?>()
+    private var latestPig: EntityPig? = null
 
     @SubscribeEvent
     fun onRenderSpecialLivingPre(event: RenderLivingEvent.Specials.Pre<EntityLivingBase?>) {
         if (!Utils.inSkyblock) return
         val e = event.entity
-        if (e !is EntityArmorStand || !e.hasCustomName() || e.isDead) return
-        if (e.customNameTag != "§6§lSHINY ORB") return
+        if (e !is EntityArmorStand || !e.hasCustomName() || e.isDead || e.customNameTag != "§6§lSHINY ORB") return
         val pos = e.position
         mc.theWorld.getEntitiesWithinAABBExcludingEntity(
             e,
             AxisAlignedBB(pos, BlockPos(pos.x + 1, pos.y + 1, pos.z + 1))
         ).find {
             if (it is EntityArmorStand && it.hasCustomName() && it.getCustomNameTag().contains(mc.thePlayer.name)) {
-                it.worldObj.removeEntity(it)
+                it.worldObj.removeEntity(it);
                 e.worldObj.removeEntity(e)
-                orbLocations.add(Vec3(pos.x + 0.5, (pos.y - 2).toDouble(), pos.z + 0.5))
+                shinyPigs.putIfAbsent(Vec3(pos.x + 0.5, (pos.y - 2).toDouble(), pos.z + 0.5), latestPig)
+                latestPig = null
                 return@find true
             }
             return@find false
         }
     }
-
     @SubscribeEvent
     fun onWorldRender(event: RenderWorldLastEvent) {
         if (!Utils.inSkyblock) return
         if (SBInfo.mode != SkyblockIsland.Hub.mode && SBInfo.mode != SkyblockIsland.FarmingIsland.mode) return
         if (!Skytils.config.shinyOrbWaypoints) return
 
+        val iterator = shinyPigs.entries.iterator();
+        while (iterator.hasNext()) {
+            val it = iterator.next();
+            if (it.value == null || it.value!!.isDead) {
+                iterator.remove()
+            }
+        }
```

Comment:
This can be replaced with a `removeIf`
```suggestion
        shinyPigs.entries.removeIf { (_, pig) ->
                pig?.isDead != false
        }
```

Reply:
`shinyPigs.values.removeIf { it?.isDead != false }` should work as well